client/collector/src/main/java/org/hawkular/apm/client/collector/internal/DefaultTraceCollector.java

```diff
@@ -47,9 +47,9 @@
 import org.hawkular.apm.api.services.TracePublisher;
 import org.hawkular.apm.api.utils.EndpointUtil;
 import org.hawkular.apm.api.utils.PropertyUtil;
-import org.hawkular.apm.client.api.reporter.BatchTraceReporter;
 import org.hawkular.apm.client.collector.SessionManager;
 import org.hawkular.apm.client.collector.TraceCollector;
+import org.hawkular.apm.trace.publisher.rest.client.BatchTraceRecorder;
```

Comment:
Yes, it should not depend on a specific impl :+1: , it was moved back to the API. Builder was modified to accept `publisher` -> `withPublisher(publisher)` rather than instantiating REST impl.